exercise/core/tests/test_recipe_api.py

```diff
@@ -0,0 +1,196 @@
+from django.test import TestCase
+from django.urls import reverse
+from core.serializers import RecipeSerializer
+
+from rest_framework import status
+from rest_framework.test import APIClient
+from core.models import Recipe, Ingredient
+from django.core.exceptions import ObjectDoesNotExist
+
+
+RECIPE_URL = reverse("recipe:recipe-list")
+
+
+def recipe_detail_url(recipe_id):
+    return reverse("recipe:recipe-detail", args=[recipe_id])
+
+
+def sample_ingredient(name, recipe):
+    return Ingredient.objects.create(name=name, recipe=recipe)
+
+
+def sample_recipe(**params):
+    default = {"name": "sample recipe", "description": "sample description"}
+    default.update(**params)
+    return Recipe.objects.create(**params)
+
+
+class RecipeTest(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_recipe_list_retrieval(self):
+        recipe_1 = sample_recipe(
+            name="French Onion Soup", description="So delicious"
+        )
+
+        sample_recipe(
+            name="Greek Salad", description="its all about the tomatoes"
+        )
+        sample_ingredient(name="Onion", recipe=recipe_1)
+        sample_ingredient(name="Water", recipe=recipe_1)
+
+        res = self.client.get(RECIPE_URL)
+
+        recipes = Recipe.objects.all()
+        serialize = RecipeSerializer(recipes, many=True)
+        self.assertEqual(len(recipes), 2)
```

Comment:
premature optimization :wink: 
```suggestion
        self.assertEqual(recipes.count(), 2)
```
In the case that recipe would be big table, this would make the query smaller since only the count is transferred and not all objects.